bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/discovery/HaperoDiscoveryService.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.discovery;
+
+import java.util.Collections;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hapero.internal.HaperoBindingConstants;
+import org.openhab.binding.hapero.internal.handler.HaperoBridgeHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link HaperoDiscoveryService} is responsible for discovery of new devices.
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = HaperoDiscoveryService.class, immediate = true, configurationPid = "discovery.hapero")
+public class HaperoDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private static final int DISCOVER_TIMEOUT_SECONDS = 5;
+    private static final String DEVICEID = "deviceID";
+
+    private @NonNullByDefault({}) HaperoBridgeHandler bridgeHandler;
+
+    /**
+     * Constructor
+     */
+    public HaperoDiscoveryService() {
+        super(Collections.singleton(new ThingTypeUID(HaperoBindingConstants.BINDING_ID, "-")), DISCOVER_TIMEOUT_SECONDS,
```

Comment:
You need to provide valid supported thing types here. You have them defined at the top of `HaperoBindingConstants`.

Reply:
I need to ask you for an example that you are going to accept here. The openhab documentation does not mention this and i looked in three or four bindings and they all seem to do it differently or only have one type, like the danfoss. I just would like to avoid having to change it a couple of times until its the "right" way.

Reply:
Like so ?
In HaperoThingTypes.java:

`    // List of all Thing Type UIDs
    public static final ThingTypeUID THING_TYPE_BRIDGE = new ThingTypeUID(BINDING_ID, "haperoBridge");
    public static final ThingTypeUID THING_TYPE_FURNACE = new ThingTypeUID(BINDING_ID, "furnace");
    public static final ThingTypeUID THING_TYPE_BUFFER = new ThingTypeUID(BINDING_ID, "buffer");
    public static final ThingTypeUID THING_TYPE_BOILER = new ThingTypeUID(BINDING_ID, "boiler");
    public static final ThingTypeUID THING_TYPE_HEATING = new ThingTypeUID(BINDING_ID, "heatingCircuit");
    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = new HashSet<>(Arrays.asList(THING_TYPE_BRIDGE, THING_TYPE_FURNACE, THING_TYPE_BUFFER, THING_TYPE_BOILER, THING_TYPE_HEATING));`

in HaperoDiscoveryService.java:

`@Activate
    public HaperoDiscoveryService(@Reference TranslationProvider i18nProvider,
            @Reference LocaleProvider localeProvider) {
        super(HaperoBindingConstants.SUPPORTED_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS, true);
        this.i18nProvider = i18nProvider;
        this.localeProvider = localeProvider;
    }`

Reply:
Exactly. I don't remember if the bridge should be included, just consider that.